src/main/webapp/assets/js/feedScript.js

```diff
@@ -226,35 +232,22 @@ function onKeyUp(event) {
   }
 }
 
-/**
- * Gets the secret value corresponding to a secret ID from GCP secrets store.
- * @param {string} secretid - The secret's id, as defined in the secrets store.
- * @return {string | null} - Either the secret for the requested ID, or else null.
- */
-async function getSecretFor(secretid) {
-  try {
-    const response = await fetch('/secretsManager?id=' + secretid, {method: 'POST'});
-    if (!response.ok) {
-      throw new Error(response.status);
-    } else {
-      return await response.text();
-    }
-  } catch (err) {
-    console.warn(err);
-    removeMapSpinner();
-    return null;
-  }
-}
-
 /**
  * Translates a location from its name to a pair of latitude and longitudes.
  * @param {string} address - The address to translate to lat/long.
  * @param {function} apiToCall - Represents the option to dependency-inject a mock api call.
+ * @param {array} locality - Represents the option to inject the current college's location.
  * @return {Promise<LocationInfo>} or null if no such location exists or an error occurs.
  */
-async function translateLocationToLatLong(address, apiToCall = fetchTranslateLocation) {
+async function translateLocationToLatLong(
+    address,
+    apiToCall = fetchTranslateLocation,
+    locality = collegeLocation) {
   try {
-    const response = await apiToCall(address);
+    // Add the college's city if it isn't already present in the query so that
+    // geolocator has a better chance at succeeding in finding lat/longs.
+    const city = locality ? locality.city : '';
+    const response = await apiToCall(address.includes(city) ? address : `${address}, ${city}`);
 
     if (!response) {
       throw new Error('POST failed for unknown reasons. Please check console.');
```

Comment:
Will the user see this? We would generally not want to expose messages to the user to check the console. You could possibly just say, it failed for unknown reasons.

In cases like this, there could be further logging added when in debug mode (I wouldn't prioritize that though).

Reply:
The user wouldn't see this directly, it would appear in the console as a warning :) If the user is already in the console, though, it's probably unnecessary to tell them to further check the console though, so I'll change it!